src/content/en/2022/privacy.md

```diff
@@ -1,22 +1,471 @@
+
+## Access to (sensitive) data from the browser
+
+Browsers have an abundant number of APIs, which provide developers with useful mechanisms to interact with different components in whichever way they want. Several of these APIs can also be used to extract information from sensors or other peripherals connected to the user's device. While most APIs provide a limited amount of information, such as the orientation of the screen, others provide very detailed information (e.g. the accelerometer and gyroscope), which could be used for device fingerprinting, or even inferring which password a user types based on the movements they make with their mobile device.
+
+### Sensor events
+
+{{ figure_markup(
+  image="sensor-events.png",
+  caption="Top device sensor events used.",
+  description="Bar chart showing which percentage of websites use certain sensor events. The event named `deviceOrientation` was found to be used on 4.04% desktop and 4.10% mobile sites, `deviceReady` on 1.16% and 1.28%, `devicemotion` on 0.78% and 0.72%, `deviceChange` on 0.29% and 0.28%, and finally the `deviceproximity` event was found on 0.03% desktop and 0.02% mobile sites.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vRSnqWDZQ07x6NdnIix7JOfg5DqGcoTrrpwaKavCIvNFkIANncyMxahKaTYjLka3rsrcBmrRIURpnPo/pubchart?oid=2114701877&format=interactive",
+  sheets_gid="217371442",
+  sql_file="number_of_websites_with_device_sensor_events.sql"
+  )
+}}
+
+We find that the most prominent sensor event that websites listen for is the `deviceOrientation` event, which fires when the device changes from portrait to landscape mode or vice versa. It is used on 4.04% of desktop websites and 4.10% of mobile websites. The usage is likely this high (relatively) because websites might want to update elements of the layout when the orientation of the device changes.
+
+### Media devices
+
+{{ figure_markup(
+  caption="The percentage of desktop pages that enumerate media devices.",
+  content="0.59%",
+  classes="big-number",
+  sheets_gid="1554147968",
+  sql_file="number_of_websites_with_device_sensor_blink_usage.sql"
+)
+}}
+
+Using the [MediaDevices API](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices), web developers can use the `enumerateDevices()` method to get a list of all media devices connected to the user's device. While this feature is useful to determine whether a user has a camera or microphone connected to initiate a video call, it can also be used to gather information about the system's environment for fingerprinting purposes. We find that 0.59% desktop websites and 0.48% mobile sites try to access the list of connected media devices (note that our crawler does not interact with the site, nor click on any buttons). Interestingly, the usage of this API has significantly reduced since last year, when **7.23%** of desktop sites were accessing the list of media devices. Most likely this is due to a popular library that no longer calls the API.
```

Comment:
```suggestion
Using the [MediaDevices API](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices), web developers can use the `enumerateDevices()` method to get a list of all media devices connected to the user's device. While this feature is useful to determine whether a user has a camera or microphone connected to initiate a video call, it can also be used to gather information about the system's environment for fingerprinting purposes. We find that 0.59% of desktop websites and 0.48% of mobile sites try to access the list of connected media devices (note that our crawler does not interact with the site, nor click on any buttons). Interestingly, the usage of this API has significantly reduced since last year, when **7.23%** of desktop sites were accessing the list of media devices. Most likely this is due to a popular library that no longer calls the API.
```